### PR TITLE
DAOS-623 cart: Remove return code for ivo_on_put.

### PR DESCRIPTION
**Pull request by ashleypittman**
Change on_put and ent_put to return void.  The return code
was never set and never read.

Required-githooks: true

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/cart/crt_iv.c [hunk 1]
```diff
@@ -2532,16 +2521,10 @@ handle_ivupdate_response(const struct crt_cb_info *cb_info)
...
 
-			if (rc != 0) {
-				D_ERROR("ivo_on_put(): "DF_RC"\n", DP_RC(rc));
-				child_output->rc = rc;
-			} else {
-				child_output->rc = output->rc;
-			}
+			child_output->rc = output->rc;
 		} else {
 			child_output->rc = output->rc;
```
Comment (frostedcmos) [inline src/cart/crt_iv.c:2529]: should just move this outside of if statement

**Review by frostedcmos — COMMENTED** [review]
small comment; looks good otherwise

**Review by soumagne — APPROVED** [review]

File: src/cart/crt_iv.c [hunk 2]
```diff
@@ -2888,9 +2867,7 @@ bulk_update_transfer_done_aux(const struct crt_bulk_cb_info *info)
 			D_GOTO(exit, rc);
 		}
 
-		output->rc = iv_ops->ivo_on_put(ivns_internal,
-						&cb_info->buc_iv_value,
-						cb_info->buc_user_priv);
+		iv_ops->ivo_on_put(ivns_internal, &cb_info->buc_iv_value, cb_info->buc_user_priv);
```
Comment (frostedcmos) [inline src/cart/crt_iv.c:2871]: output->rc might end up being unset in this branch, need to set it to DER_SUCCESS either here or after line 2800 to be safe

Comment (ashleypittman) [reply src/cart/crt_iv.c:2871]: I did wonder about that and had that logic in a earlier version of this PR but DER_SUCCCESS is zero and the memory should be zeroed already?  Unless there's something that's already there that needs to be overwritten.

File: src/include/daos_srv/iv.h [hunk 3]
```diff
@@ -164,9 +164,8 @@ typedef int (*ds_iv_ent_get_t)(struct ds_iv_entry *ent, void **priv);
  * \param ent [IN]	iv class entry to get.
  * \param priv [IN]	private ptr from crt IV callback.
  *
- * \return		0 if succeeds, error code otherwise.
  */
-typedef int (*ds_iv_ent_put_t)(struct ds_iv_entry *ent, void *priv);
+typedef void (*ds_iv_ent_put_t)(struct ds_iv_entry *ent, void *priv);
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/include/daos_srv/iv.h:168]: I believe so.  We should push back hard against making our internal headers backwards compatible.

File: src/cart/crt_iv.c (hunk 2 above)
Comment (ashleypittman) [reply src/cart/crt_iv.c:2871]: Done.

File: src/include/daos_srv/iv.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (frostedcmos) [reply src/include/daos_srv/iv.h:168]: > Only `src/include/*.h` needs to be backwards compatible, right? Not also `src/include/*/*.h`?

both locations include public header files, e.g. src/include/cart/api.h, so you cant make a distinction based on the path. 

Seems to me that src/include/daos_srv/  might be misplaced as the rest of the files under src/include/ appear to be public headers

**Review by frostedcmos — APPROVED** [review]

File: src/include/daos_srv/iv.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/include/daos_srv/iv.h:168]: yeah, our headers and includes are a mess.  I think we're fine to modify the `daos_srv` ones though.

[the remark this answers is not in the document]
Comment (daltonbohning) [reply src/include/daos_srv/iv.h:168]: FWIW, in my dev build
```
$ ls /home/dbohning/daos/install/include/daos_srv
vos.h  vos_types.h
```
So this specific file is okay

**Review by daltonbohning — APPROVED** [review]

File: src/include/daos_srv/iv.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/include/daos_srv/iv.h:168]: Like I say, our headers are a mess.

**Pull request merged**